nettest/src/glue.rs

```diff
@@ -0,0 +1,206 @@
+            DemikernelSyscall::Unsupported => write!(f, "Unsupported"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(non_camel_case_types)]
+pub enum SocketDomain {
+    AF_INET,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(non_camel_case_types)]
+pub enum SocketType {
+    SOCK_STREAM,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(non_camel_case_types)]
+pub enum SocketProtocol {
+    IPPROTO_TCP,
+}
+
+#[derive(Clone, Debug)]
+pub struct SocketArgs {
+    pub domain: SocketDomain,
+    pub typ: SocketType,
+    pub protocol: SocketProtocol,
+}
+
+#[derive(Clone, Debug)]
+pub struct BindArgs {
+    pub qd: Option<u32>,
+    pub addr: Option<SocketAddr>,
+}
+
+#[derive(Clone, Debug)]
+pub struct ListenArgs {
+    pub qd: Option<u32>,
+    pub backlog: Option<usize>,
+}
+
+#[derive(Clone, Debug)]
+pub struct AcceptArgs {
+    pub qd: Option<u32>,
+}
+
+#[derive(Clone, Debug)]
+pub struct ConnectArgs {
+    pub qd: Option<u32>,
+    pub addr: Option<SocketAddr>,
+}
+
+#[derive(Clone, Debug)]
+pub struct PushArgs {
+    pub qd: Option<u32>,
+    pub buf: Option<Vec<u8>>,
+    pub len: Option<u32>,
+}
+
+#[derive(Clone, Debug)]
+pub enum PacketEvent {
+    Tcp(PacketDirection, TcpPacket),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum PacketDirection {
+    Incoming,
+    Outgoing,
+}
+
+#[derive(Clone, Debug)]
+pub struct TcpPacket {
+    pub flags: TcpFlags,
+    pub seqnum: TcpSequenceNumber,
+    pub ack: Option<u32>,
+    pub win: Option<u32>,
+    pub options: Vec<TcpOption>,
+}
+#[derive(Clone)]
+pub struct TcpFlags {
+    pub syn: bool,
+    pub ack: bool,
+    pub fin: bool,
+    pub rst: bool,
+    pub psh: bool,
+    pub urg: bool,
+    pub ece: bool,
+    pub cwr: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct TcpSequenceNumber {
+    pub seq: u32,
+    pub ack: u32,
+    pub win: u32,
+}
+
+#[derive(Clone, Debug)]
+pub enum TcpOption {
+    Noop,
+    Mss(u16),
+    WindowScale(u8),
+    SackOk,
+    Timestamp(u32, u32),
+    EndOfOptions,
+}
+
+impl Debug for TcpFlags {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut flags = String::new();
+        if self.syn {
+            flags.push_str("SYN ");
+        }
+        if self.ack {
+            flags.push_str("ACK ");
+        }
+        if self.fin {
+            flags.push_str("FIN ");
+        }
+        if self.rst {
+            flags.push_str("RST ");
+        }
+        if self.psh {
+            flags.push_str("PSH ");
+        }
+        if self.urg {
+            flags.push_str("URG ");
+        }
+        if self.ece {
+            flags.push_str("ECE ");
+        }
+        if self.cwr {
+            flags.push_str("CWR ");
+        }
+        write!(f, "{}", flags)
+    }
+}
+
+pub fn parse_int(s: &str) -> Result<u32, ()> {
+    match s.parse::<u32>() {
+        Ok(val) => Ok(val),
+        Err(_) => {
+            eprintln!("{} cannot be represented as a u64", s);
```

Comment:
Should the err msg refer to u32? Maybe I'm not reading the intent correctly here.